cmd/transportRequestUploadSOLMAN.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/transportrequest/solman"
+)
+
+type transportRequestUploadSOLMANUtils interface {
```

Comment:
> Why dont you use the common.go Exec interface?

Would be possible technically. We had some discussion and still no final conclusion how to deal with that. In go it seems (?) to be idiomatic to define the interfaces more in a per-usecase level and to "trust" that the runtime instance fullfills the interface (if not we will get some error message of course).

> I like your "separation of concerns".

Don't get the point.

> move the FileSystem interface into common.go 

The file system interface is defined [here](https://github.com/SAP/jenkins-library/blob/master/pkg/piperutils/FileUtils.go#L17). Of course we can more the interfaces around, but what is wrong with the current location?

> and reuse it.

We can't we re-use it from the current location?

Reply:
thought about something like
```
type solmanUploader struct {
  command solman.Exec  // from solman/common.go
  fs solman.FileSystem // from solman/common.go
}
...
env := solmanUploader{
	Command: &command.Command{},
	FS:      &piperutils.Files{},
}
env.Command.Stdout(log.Writer())
env.Command.Stderr(log.Writer())
...
err := action.Perform(env.FS, env.Command)
```

wouldnt need `type transportRequestUploadSOLMANUtils interface {...`

Reply:
Maybe we can discuss this in a bigger round since that affects the generator  (`type transportRequestUploadSOLMANUtils interface {...` is generated ...)

Reply:
`transportRequestUploadSOLMANUtils  `generated, are you sure? looks like a local item.

Reply:
> `transportRequestUploadSOLMANUtils` generated, are you sure? looks like a local item.

[yes](https://github.com/SAP/jenkins-library/blob/484d9c2ec113958e8e7d86e49df1f38419946385/pkg/generator/helper/helper.go#L264). Also local items can be generated ...

Reply:
But transportRequestUploadSOLMANUtils  isnt used anywhere but in transportRequestUploadSOLMAN.go. It is neither defined nor used within a _generated.go file

Reply:
What should I say?

I added a dummy step here and created the corresponding file via`go generate`. I checked in the files as they has been generated. Please have a look here:

- https://github.com/marcusholl/jenkins-library/blob/tmp/dummyStep/cmd/transportRequestUploadCTSXXX.go
- https://github.com/marcusholl/jenkins-library/blob/tmp/dummyStep/cmd/transportRequestUploadCTSXXX_generated.go
- maybe see also the generated test files (not that important here ...)

Here we can find the entities we are talking about:
- [interface](https://github.com/marcusholl/jenkins-library/blob/5771c68192a86b9809e2e284e4fb86f5e661b710/cmd/transportRequestUploadCTSXXX.go#L11) 
- [impl](https://github.com/marcusholl/jenkins-library/blob/5771c68192a86b9809e2e284e4fb86f5e661b710/cmd/transportRequestUploadCTSXXX.go#L22)
- [factory method](https://github.com/marcusholl/jenkins-library/blob/5771c68192a86b9809e2e284e4fb86f5e661b710/cmd/transportRequestUploadCTSXXX.go#L32)

We have also some generated coding outside `*_generated`. Maybe that is causing some confusion here.

Reply:
i just learnd [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s).